python/ray/experimental/iter.py

```diff
@@ -0,0 +1,556 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from typing import TypeVar, Generic, Iterable, List, Callable, Any
+
+import ray
+
+# The type of an iterator element.
+T = TypeVar("T")
+
+
+class YieldIterator(Exception):
+    """Indicates that a local iterator has no value available.
+
+    This is used internally to implement the union() of multiple blocking
+    local generators."""
+    pass
+
+
+def from_items(items: List[T], num_shards: int = 2) -> "ParIterator[T]":
```

Comment:
nit, but I much prefer `ParallelIterator` to `ParIterator` even if it's more verbose